Setup.hs

```diff
@@ -15,22 +16,46 @@ myBuild pd lbi uh flags = do
   let dir = autogenPackageModulesDir lbi
   createDirectoryIfMissing True dir
 
-  hasGit <- findExecutable "git"
+  desc <- withExe "." "git" "<VCS-less build>" "<non-dev-build>" $ \git ->
+    init <$> readProcess git ["describe", "--always", "--dirty"] ""
 
-  let gitfailure :: SomeException -> IO String
-      gitfailure _e = return "<non-dev-build> "
+  aig_desc <- withExe "deps/aig" "git" Nothing Nothing $ \git -> do
+    Just . init <$> readProcess git ["describe", "--always", "--dirty"] ""
+                  
+  w4_desc <- withExe "deps/what4" "git" Nothing Nothing $ \git -> do
+    Just . init <$> readProcess git ["describe", "--always", "--dirty"] ""
 
-  desc <- case hasGit of
-            Just git -> readProcess "git" ["describe", "--always", "--dirty"] ""
-                        `catch` gitfailure
-            Nothing -> return "<VCS-less build> "
+  sbv_ver <- withExe "." "cabal" Nothing Nothing $ \cabal -> do
+    ex <- doesFileExist "cabal.project.freeze"
+    unless ex $ callProcess cabal ["freeze"]
+    wss <- fmap words <$> lines <$> readFile "cabal.project.freeze"
+    unless ex $ removeFile "cabal.project.freeze"
+    case find (\ws -> (ws !! 0) == "any.sbv") wss of
+      Just ws -> return . Just . init . drop 2 $ ws !! 1
+      Nothing -> return Nothing
 
   writeFile (dir </> "GitRev.hs") $ unlines
     [ "module GitRev where"
     , "hash :: String"
-    , "hash = " ++ show (init desc)
+    , "hash = " ++ show desc
```

Comment:
Perhaps it's worth leaving a comment about what kind of `hash` in particular this is, since there are now multiple types of hashes being written here.